x/fees/keeper/keeper.go

```diff
@@ -0,0 +1,69 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/params"
+	"github.com/desmos-labs/desmos/x/fees/types"
+)
+
+type Keeper struct {
+	// The reference to the ParamsStore to get and set params
+	paramSubspace params.Subspace
+	Cdc           *codec.Codec
+}
+
+func NewKeeper(cdc *codec.Codec, paramSpace params.Subspace) Keeper {
+	if !paramSpace.HasKeyTable() {
+		paramSpace = paramSpace.WithKeyTable(types.ParamKeyTable())
+	}
+
+	return Keeper{
+		Cdc:           cdc,
+		paramSubspace: paramSpace,
+	}
+}
+
+// SetParams sets params on the store
+func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
+	k.paramSubspace.SetParamSet(ctx, &params)
+}
+
+// GetParams returns the params from the store
+func (k Keeper) GetParams(ctx sdk.Context) (p types.Params) {
+	k.paramSubspace.GetParamSet(ctx, &p)
+	return p
+}
+
+// CheckFees checks whether the given fees are sufficient to pay for all the given messages.
+// The check is performed considering the minimum fee amounts specified inside the module parameters.
+func (k Keeper) CheckFees(ctx sdk.Context, fees authtypes.StdFee, msgs []sdk.Msg) error {
+	feesParams := k.GetParams(ctx)
```

Comment:
I think that we should check if the params are `nil` or not. Not having them set previously, this would return `nil` and then the access to its field would panic the whole application. 

I think that if no params are set, we should return here without an error (since there are indeed no min fees to be paid), instead of continuing the execution.

Also, a test case should be added to make sure that this particular case does not interrupt the code execution

Reply:
I think that we could do it better by editing the `InitGenesis()` function and set the default params (so basically zero fees) whenever the passed `params` from genesis are empty. What do you think?

Reply:
I guess that wouldn't solve the problem. Let me explain it better. 

If the `fees` object is not set inside the genesis, the module will not even be initialized. This means that `InitGenesis` and all the other methods inside `module.go` will not be called. 

The problem is that, even thought the module is not going to be initialized, the `AnteHandler` is still going to be called if it is set. This will cause the `k.GetParams` to return `nil` (since they have never been initialized) and thus error. 

There are two alternatives in this case: 

1. We handle that error and return as soon as we identify that the params are `nil`:  
   ```go
   feesParams := k.GetParams(ctx)
   if feesParams == nil {
      return nil
   }
   ```

2. We return an error telling what's wrong:  
   ```go
   feesParams := k.GetParams(ctx)
   if feesParams == nil {
      return fmt.ErrorF("fees params cannot be null. Please make sure you have initialized the module correctly")
   } 
   ```

I guess that the best approach is number 2. This would allow the developer to know what's wrong and fix it accordingly.